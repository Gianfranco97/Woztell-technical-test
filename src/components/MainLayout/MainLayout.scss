.layout {
  height: 100vh;

  &__main {
    overflow: auto;
    position: relative;
    margin-left: auto;
    margin-right: auto;
    padding-right: 15px;
    padding-left: 15px;

    @media (min-width: 476px) {
      padding-right: 15px;
      padding-left: 15px;
      width: 100%;
    }

    @media (min-width: 768px) {
      padding-right: 15px;
      padding-left: 15px;
      width: 720px;
      max-width: 100%;
    }

    @media (min-width: 992px) {
      padding-right: 15px;
      padding-left: 15px;
      width: 960px;
      max-width: 100%;
    }

    @media (min-width: 1200px) {
      padding-right: 15px;
      padding-left: 15px;
      width: 1140px;
      max-width: 100%;
    }
  }

  &__header {
    h1 {
      color: white;
    }
  }

  &__footer {
    border-top: 1px solid #d3d3d3;
  }

  &__footer,
  &__header {
    text-align: center;
  }
}
