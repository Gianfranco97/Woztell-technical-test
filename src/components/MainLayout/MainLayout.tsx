import { Layout } from 'antd';
import "./MainLayout.scss"

type DashboardLayoutProps = {
  children: React.ReactNode,
};

export default function MainLayout({ children }: DashboardLayoutProps) {
  return (
    <Layout className="layout">
      <Layout.Header className="layout__header"><h1>Technical test - Woztell</h1></Layout.Header>

      <Layout>
        <Layout.Content className="layout__main">{children}</Layout.Content>
      </Layout>

      <Layout.Footer className="layout__footer">Â© 2022 - Gianfranco Manganiello</Layout.Footer>
    </Layout>
  )
}