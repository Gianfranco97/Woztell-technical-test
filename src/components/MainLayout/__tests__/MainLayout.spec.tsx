import { expect } from '@jest/globals';
import { render } from '@testing-library/react'
import MainLayout from '../index';

it('The snapshot matches', () => {
  const { container } = render(
    <MainLayout><h1>Hollo world</h1></MainLayout>
  )
  expect(container).toMatchSnapshot()
});
