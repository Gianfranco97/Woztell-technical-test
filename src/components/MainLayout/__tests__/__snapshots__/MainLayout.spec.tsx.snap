// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The snapshot matches 1`] = `
<div>
  <section
    class="ant-layout layout"
  >
    <header
      class="ant-layout-header layout__header"
    >
      <h1>
        Technical test - Woztell
      </h1>
    </header>
    <section
      class="ant-layout"
    >
      <main
        class="ant-layout-content layout__main"
      >
        <h1>
          Hollo world
        </h1>
      </main>
    </section>
    <footer
      class="ant-layout-footer layout__footer"
    >
      Â© 2022 - Gianfranco Manganiello
    </footer>
  </section>
</div>
`;
